### PR TITLE
feat: add durable tests

### PR DESCRIPTION
**Pull request by smrz2001**

### CODE REVIEW

File: .github/workflows/durable.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    runs-on: ubuntu-latest
+    needs:
+      - publish-suite
+    strategy:
+      fail-fast: false
+      matrix:
+        network: [dev, qa, tnet, prod]
+        include:
+          - network: dev
+            composedb_urls: DEV_COMPOSEDB_URLS
```
Comment (nathanielc) [inline .github/workflows/durable.yml:69]: Where are the `DEV_COMPOSEDB_URLS` and related env vars defined?

File: Makefile [hunk 2]
```diff
@@ -89,6 +89,10 @@ hermetic-tests:
 		--clean-up \
 		--test-image ${TEST_SUITE_IMAGE}
 
+.PHONY: run-suite
```
Comment (nathanielc) [inline Makefile:92]: This is for durable tests correct? Should we rename this to `durable-tests`? Also how do you choose the environment to target? Should we a make var called DURABLE_ENV or something?

File: ci-scripts/run_suite.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (nathanielc) [inline ci-scripts/run_suite.sh:1]: Similar request here, can we rename this file to `durable-driver.sh` or similar to make it clear this is for durable tests?

**Review by nathanielc — CHANGES_REQUESTED** [review]

File: .github/workflows/durable.yml (hunk 1 above)
Comment (smrz2001) [reply .github/workflows/durable.yml:69]: In the repo secrets.

Comment (nathanielc) [reply .github/workflows/durable.yml:69]: ok, do they need to be secret? I'd like the ability to run the durable tests from a local checkout of the repo. That way we can easily trigger them to run and debug any issues. Thoughts?

Comment (smrz2001) [reply .github/workflows/durable.yml:69]: Hmm, traditionally, we've kept them in secrets, but they're pretty much our worst-kept secret. These URLs were never fully secret to begin with (they're public, just not advertised) and the DNS names have always been exposed in the swarm [peerlist](https://github.com/ceramicnetwork/peerlist/) anyway.

I'd be fine putting them in the code.

File: Makefile (hunk 2 above)
Comment (smrz2001) [reply Makefile:92]: Yeah, not very different from hermetic, except that they can be run wherever there's Docker running. Kind of like a `docker-compose` variant to the hermetic tests, which will only run in k8s.

I'll change the script to run only against durable infrastructure.

File: durable/durable-driver.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  *)
+    echo "Usage: $0 {dev|qa|tnet|prod}"
+    exit 1
+    ;;
+esac
+
+AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID-.}
+AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY-.}
+AWS_REGION=${AWS_REGION-us-east-1}
+JEST_ARGS=${JEST_ARGS---reporters=default --detectOpenHandles}
```
Comment (nathanielc) [inline durable/durable-driver.sh:39]: Why do we need the `detectOpenHandles` flag? I thought that was for debugging tests that leave running async tasks?

**Review by nathanielc — APPROVED** [review]
LGTM with one question below

File: durable/durable-driver.sh (hunk 4 above)
Comment (smrz2001) [reply durable/durable-driver.sh:39]: Hmm, good question. I stopped paying attention to the options themselves and have been copying them over the way they've been from the start. I'll remove this option.

**Pull request merged**